src/main/java/taxi/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java

```diff
@@ -0,0 +1,133 @@
+import taxi.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoJdbcImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers "
+                + "(manufacturer_name, manufacturer_country)"
+                + " VALUES(?, ?);";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement preparedStatement = connection
+                        .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject(1, Long.class));
+            }
+            resultSet.close();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not add new manufacturer "
+                    + manufacturer.getName(), e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String query = "SELECT * FROM manufacturers "
+                + "WHERE manufacturer_id = ? "
+                + "AND deleted = FALSE;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                Optional<Manufacturer> newManufacturer = Optional
+                        .of(createManufacturer(resultSet));
+                resultSet.close();
+                return newManufacturer;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not find manufacturer with id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String query = "SELECT * FROM manufacturers WHERE deleted = FALSE;";
+        List<Manufacturer> manufacturerList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                manufacturerList.add(createManufacturer(resultSet));
+            }
+            resultSet.close();
+        } catch (SQLException e) {
+            throw new DataProcessingException("No manufacturers found", e);
+        }
+        return manufacturerList;
+    }
+
+    private Manufacturer createManufacturer(ResultSet resultSet) throws SQLException {
```

Comment:
```suggestion
    private Manufacturer createManufacturer(ResultSet resultSet) {
```
if we handle exception inside of the method with a try catch construction we don't need to use throws as well

Reply:
Removed `throws` from the method signature.